backend/python/app/services/implementations/user_service.py

```diff
@@ -275,6 +278,36 @@ def update_user_by_id(self, user_id, user):
 
         return UserDTO(user_id, user.first_name, user.last_name, user.email, user.role)
 
+    def soft_delete_user(self, user_id):
+        try:
+            user = User.query.get(user_id)
```

Comment:
nit: I dont think users should be allowed to delete themselves, can you throw an exception if the user is trying to delete itself?

Reply:
Fixed!